processor/src/main/java/org/mapstruct/ap/internal/model/MethodReference.java

```diff
@@ -387,9 +387,6 @@ public boolean equals(Object obj) {
         if ( this == obj ) {
             return true;
         }
-        if ( !super.equals( obj ) ) {
-            return false;
-        }
```

Comment:
This evaluated against `Object#equals` and therefore was always false. This might be a relic from legacy inheritance structures or never worked as intended.